### PR TITLE
updates link quality preview

### PR DESCRIPTION
**Pull request by JyeSmith**
LQ preview update

### CODE REVIEW

File: src/js/tabs/osd.js [hunk 1]
```diff
@@ -1013,7 +1013,7 @@ OSD.loadDisplayFields = function() {
             default_position: -1,
             draw_order: 390,
             positionable: true,
-            preview: FONT.symbol(SYM.LINK_QUALITY) + '8'
+            preview: FONT.symbol(SYM.LINK_QUALITY) + '4:100'
```
Comment (McGiverGim) [inline src/js/tabs/osd.js]: This does not seem correct. According to the code the second parameter has a length of two:

https://github.com/betaflight/betaflight/blob/fcef76b82f785167574aed7f1019c016137ce67d/src/main/osd/osd_elements.c#L945-L959

So it will more of the type `4:99`

And this is only with crossfire. With others protocols it only has one digit, the first. I don't know if we have here the info to know if the user has configured crossfire or not because the size of the element is very different from one type to the other.

Comment (McGiverGim) [reply src/js/tabs/osd.js]: I've been looking at the code and it seems we don't have here the serial protocol selected.

I don't have crossfire. I usually add the Link Quality just side by side with the RSSI element, something like `L9R99` being L and R the symbol representation of Link Preview and RSSI, but now with this preview this is impossible to do. The RSSI is drawn before so the Link Quality preview overwrites it.

I think we have two solutions for this:
- Send the MSP command to get the serial protocol selected and draw the correct preview
- Let the old preview and the users of crossfire will have a not accurate preview.

@mikeller what do you think about this?

Comment (JyeSmith) [reply src/js/tabs/osd.js]: LQ can currently show 100, and to my way of thinking that is correct (e.g. receiving 100% of the packets).

Not sure the correct way to resolve the issue with other protocol and a single digit.

Comment (mikeller) [reply src/js/tabs/osd.js]: If we wanted to show the CRSF specific version of this element, we will have to read the selected RX protocol through MSP, and then only show the CRSF version if this is the protocol selected, and what we have now otherwise.
This is a lot of effort for something that is just a placeholder to give users an easy way to set up their OSD, so my hunch is to not go this way.

**Comment by wvarty** [comment]
Without doing the work to pull in the currently selected RX protocol, this preview will always be incorrect for one group of users (either CRSF or non-CRSF). Now I could be wrong, but I think the majority of users that put LQ on their OSD use CRSF, so would it not make sense to have the preview correct for the majority rather than the minority in the short term, and look at doing the work to update the preview dynamically in the longer term?

**Comment by McGiverGim** [comment]
The problem is that if you make the Crossfire preview as default, you use a lot more space than the rest of users need, making impossible adjust two elements in the screen one next to the other and see the marker of the elements in the Configurator. 
If you are a crossfire user, you can put one element and the other with enough space in the Configurator, and the only problem is that the preview in the Configurator will not be correct but you can see it correct in the goggles.
Maybe we are a minority, but I use the LQ element and I'm not a Crossfire user.

**Comment by etracer65** [comment]
The LQ for non-CRSF (currently only Frsky) is really a fake statistic and not very useful. It's just a calculation of a ratio of how many packets were "lost" over a very short time period. Not really very useful in predicting when signal loss will happen. Frankly having this LQ for non-CRSF is something we never should have added as it's misinformation.

**Comment by McGiverGim** [comment]
We can remove the element for non crossfire users in firmware, and then use only it here, or if we have both forms we must support both here, and the best way is use the short form.

**Comment by wvarty** [comment]
Any chance we can get some movement on this...? It's frustrating trying to guage how close I can put the LQ element to the edge of the OSD when using the new format in 4.2...

**Comment by McGiverGim** [comment]
To me, if the LQ is not useful for non crossfire, I think is a good idea to remove it. @mikeller @etracer65 what are your thoughts about this?

If we remove it, to me is ok to modify the Configurator to show the Crossfire version. 

If not I continue thinking we need to use the longer form of the shorter one to let the user situate the elements without overlapping in the interface and getting the maximum space. You can use the goggles in live while doing that if you want a live preview.

**Comment by mikeller** [comment]
@wvarty: Not sure what is so hard about positioning this element 2 extra characters away from the next element?
There are a lot of other elements for which the OSD preview does not show the maximum size of the element, and I am not aware of any other user ever complaining that they are struggling with setting up these.

**Comment by wvarty** [comment]
@mikeller sure, it’s not difficult for me or you to understand that we need to position the element 2 chars to the left. We are both aware of the merged PR that changed the length, so we understand what needs to be done. But have you had a look in the xfire support FB group lately? Lots of questions about why the LQ value “looks cutoff” now because ppl have it close to the edge of the screen. Like I originally said, majority of LQ users are xfire users.

**Comment by mikeller** [comment]
@wvarty: 

> Like I originally said, majority of LQ users are xfire users.

Do you have any data to back up this claim or are you just making things up as you go?

**Comment by wvarty** [comment]
@mikeller that actually brings up a good point: changes like this that affect a large number of users should be put to a poll on the BF Facebook group / the Slack channel. That way we can make a choice based on user requirements instead of arguing on github.

**Comment by mikeller** [comment]
@wvarty: If you want to use a firmware that makes design decisions by poll, can I recommend you start writing your own?

**Comment by wvarty** [comment]
@mikeller I thought you wanted data on the number of users who use crsf LQ vs other protocols...?

**Comment by wvarty** [comment]
bf

Here is one particular example of a user running into issues due to this...

**Comment by etracer65** [comment]
> To me, if the LQ is not useful for non crossfire, I think is a good idea to remove it. @mikeller @etracer65 what are your thoughts about this?

LQ for non-CRSF was originally added by a Futaba user for cases with Futaba SBUS receivers that had no capability for RSSI. It's really a hack that provides very little (or misleading) information about when you're about to lose signal, but I guess it's "better than nothing" (but just barely) if you don't have RSSI.

However any FrSky user with actual RSSI should never use this. RSSI is a far better and more reliable indicator. Since this "fake" LQ is based on the "dropped" bit in the SBUS frame we don't actually know what this means and it's dependent on the proprietary implementation. For example we don't know if the receiver detected a corrupted or incomplete OTA packet, maybe detected a "missing" packet based on expected arrival, etc. We just don't know. Also we have no way of knowing if Futaba's implementation or meaning is the same as FrSky's. And to muddy it even further, there are Flysky receivers that can output SBUS and now DJI's HD implementation. Again both of which we have no idea about the underlying implementation. The "dropped" bit is not meant for determining LQ so using it for this without understanding the underlying details is a bad idea.

I guess if we stopped trying to call this "link quality" and had it be a separate element to indicate "dropped" packet % or something it would make more sense. But calling it link quality is misleading.

So to summarize - if you have real RSSI never use this fake non-CRSF link quality.

**Comment by etracer65** [comment]
I'm not sure why there's such a detailed discussion about the size of the the element preview. Just make it the largest possible size and be done with it. It has nothing to do with which is "more popular". If people complain that it makes them "waste space" then too bad. They can use the live preview in their goggles. We have lots of other elements where the actual OSD display might be smaller than the preview - like all the name fields or even the warnings element.

**Comment by McGiverGim** [comment]
The problem making it larger is that if someone, like me, without crossfire adjusts it to other element, in the Configurator the second does not appear because one overwrites the other, making it difficult to adjust.

**Comment by etracer65** [comment]
> The problem making it larger is that if someone, like me, without crossfire adjusts it to other element, in the Configurator the second does not appear because one overwrites the other, making it difficult to adjust.

Or if it's the smaller version then people can put it too close to another element (or edge of the screen) and have it cutoff. So there's no way to make everybody happy. Until we introduce some kind of "outline" showing the full element size for all elements I don't see how we can resolve. Like the Warnings element previews as "LOW VOLTAGE" but could display many different things that are shorter. Or Display Name previews "JOE PILOT" but actually can be up to 16 characters. And so on.

**Comment by mikeller** [comment]
@etracer65: The way the OSD preview is currently implemented the better approach is to make preview elements match their smallest form - since the preview will not allow users to position elements in a way that they are clipping the screen boundary, users that use a smaller version of the element will be forced to waste screen space by positioning their element at some distance from the side.
